crates/deltalake-aws/src/lib.rs

```diff
@@ -266,16 +288,37 @@ impl DynamoDbLockClient {
             ..Default::default()
         };
 
-        match self.dynamodb_client.update_item(input).await {
-            Ok(_) => Ok(UpdateLogEntryResult::UpdatePerformed),
-            Err(RusotoError::Service(UpdateItemError::ConditionalCheckFailed(_))) => {
-                Ok(UpdateLogEntryResult::AlreadyCompleted)
+        retry(|| async {
+            match self.dynamodb_client.update_item(input.clone()).await {
+                Ok(_) => Ok(UpdateLogEntryResult::UpdatePerformed),
+                Err(RusotoError::Service(UpdateItemError::ConditionalCheckFailed(_))) => {
+                    Ok(UpdateLogEntryResult::AlreadyCompleted)
+                }
+                Err(RusotoError::Service(UpdateItemError::ProvisionedThroughputExceeded(_))) => {
+                    Err(backoff::Error::transient(
+                        LockClientError::ProvisionedThroughputExceeded,
+                    ))
+                }
+                Err(err) => Err(backoff::Error::permanent(err.into())),
             }
-            Err(err) => Err(err)?,
-        }
+        })
+        .await
     }
 }
 
+async fn retry<I, E, Fn, Fut>(operation: Fn) -> Result<I, E>
+where
+    Fn: FnMut() -> Fut,
+    Fut: std::future::Future<Output = Result<I, backoff::Error<E>>>,
+{
+    let backoff = backoff::ExponentialBackoffBuilder::new()
+        .with_multiplier(2.)
+        .with_max_interval(Duration::from_secs(15))
+        .with_max_elapsed_time(Some(Duration::from_secs(60)))
```

Comment:
The max elapsed interval of 60 I am assuming is just a number that you've picked? 

I think this is something that will likely need to be configurable since different systems would have a different tolerance here. Based on how `retry` is being invoked I don't see a clear path to pulling configuration down into this call :thinking:

Reply:
Yes, these numbers would ideally be configurable. It seems other aspects of `delta-rs`, including the locking provider choice itself, are being configured via environment variables, so I guess that's the way to go.

My question would mostly be if we want all five of possible parameters to be configurable:
- max elapsed time
- max interval
- multiplier
- initial interval (using defaults in this PR)
- randomization factor (using defaults in this PR)

The most obvious one is probably the one you singled out, max elapsed time, but there's probably a user out there for any of these :).

Reply:
@dispanser I agree there are folks likely that would want to tune every single one of these...maybe.

I think making an environment variable for max elapsed time is the only one really important to implement to merge this. The others people may want, or not, but they should make themselves known in the future :smile:

Reply:
I added an environment variable for the max elapsed time setting.

I'd address the diverging docs in a separate PR, they are already way off anyways :)